delta/service/src/main/scala/ch/epfl/bluebrain/nexus/delta/service/organizations/OrganizationsImpl.scala

```
@@ -0,0 +1,257 @@
+    else {
+      eventLog
+        .currentEventsByPersistenceId(s"$entityType-$label", Long.MinValue, Long.MaxValue)
+        .takeWhile(_.event.rev <= rev)
+        .fold[OrganizationState](OrganizationState.Initial) {
+          case (state, event) => Organizations.next(state, event.event)
+        }
+        .compile
+        .last
+        .hideErrors
+        .flatMap {
+          case Some(state) if state.rev == rev => UIO.pure(state.toResource)
+          case Some(_)                         =>
+            fetch(label).flatMap {
+              case Some(res) => IO.raiseError(RevisionNotFound(rev, res.rev))
+              case None      => IO.pure(None)
+            }
+          case None                            => IO.raiseError(RevisionNotFound(rev, 0L))
+        }
+        .named("fetchOrganizationAt", component)
+    }
+
+  override def fetch(uuid: UUID): UIO[Option[OrganizationResource]] =
+    fetchFromCache(uuid)
+      .flatMap {
+        case Some(label) => fetch(label)
+        case None        => UIO.pure(None)
+      }
+      .named("fetchOrganizationByUuid", component)
+
+  override def fetchAt(uuid: UUID, rev: Long): IO[RevisionNotFound, Option[OrganizationResource]] =
+    fetchFromCache(uuid)
+      .flatMap {
+        case Some(label) => fetchAt(label, rev)
+        case None        => UIO.pure(None)
+      }
+      .named("fetchOrganizationAtByUuid", component)
+
+  private def fetchFromCache(uuid: UUID): UIO[Option[Label]]                                  =
+    cache.collectFirst { case (label, resource) if resource.value.uuid == uuid => label }
+
+  private def eval(cmd: OrganizationCommand): IO[OrganizationRejection, OrganizationResource] =
+    for {
+      evaluationResult <- agg.evaluate(cmd.label.value, cmd).mapError(_.value)
+      resource         <- IO.fromOption(evaluationResult.state.toResource, UnexpectedInitialState(cmd.label))
+      _                <- cache.put(cmd.label, resource)
```

I think that maybe we should ignore errors for the cache update, as it would be recovered by the async index, so something like: `cache.put(cmd.label, resource).as(()).recoverWith(_ => IO.unit)`

---

cache.put returns an `UIO`. There is nothing to recover from it.